### PR TITLE
internal errors don't include calls

### PR DESCRIPTION
**Pull request by romainfrancois**
``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

# through Rcpp_eval
mutate(tibble(x=1:10) %>% mutate(z=y))
#> Error: Evaluation error: object 'y' not found.

# through stop
n_distinct()
#> Error: Need at least one column for `n_distinct()`
```

Created on 2018-06-18 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

### CODE REVIEW

**Comment by romainfrancois** [comment]
We can either do it in the `Makevars` files as: 

```
-DRCPP_DEFAULT_INCLUDE_CALL=false
```

or in the code by adding 

```
#define RCPP_DEFAULT_INCLUDE_CALL false
```

in `dplyr.h`

**Comment by romainfrancois** [comment]
Not sure what the travis error is about. https://travis-ci.org/tidyverse/dplyr/jobs/393536260#L944

image

File: tests/testthat/test-mutate.r [hunk 1]
```diff
@@ -820,3 +820,21 @@ test_that("grouped subsets are not lazy (#3360)", {
 
   expect_identical(res, list(make_call("a"), make_call("b")))
 })
+
+test_that("errors don't have tracebacks (#3662)", {
+  tryCatch(
```
Comment (hadley) [inline tests/testthat/test-mutate.r]: I think it's little nicer to write this as:

```
err <- capture_condition(mutate(tibble(x = 1:10) %>% mutate(z = y)))
```

I'm also worried about specific error messages as these will be sensitive to unrelated changes. 

(And can you please use styler to apply tidyverse style here)

**Comment by romainfrancois** [comment]
Bumped it to Rcpp 0.12.18 which is the current CRAN version and applied @hadley's suggestion of using capture_condition. Should be good to go.
